src/Npgsql/NpgsqlException.cs

```diff
@@ -50,6 +51,23 @@ public NpgsqlException(string message)
         public virtual bool IsTransient =>
             InnerException is IOException || InnerException is SocketException;
 
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            var builder = new StringBuilder(base.ToString());
+            builder.AppendLine();
+            builder.AppendLine("Exception data:");
```

Comment:
`AppendLine` returns the `StringBuilder` on which the method is called. So you can chain invocations.